### PR TITLE
[sailfish-browser] Add PopupBlockerDialog. Contributes to JB#52731

### PR DESCRIPTION
**Pull request by atatarov**
We want to notify the user when pop-ups are blocked.
Handling embed:popupblocked allows to get these events.

### CODE REVIEW

File: apps/shared/PopupBlockedDialog.qml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            //% "Deny"
+            cancelText: qsTrId("sailfish_browser-he-block_popup")
+        }
+
+        Label {
+            x: Theme.horizontalPageMargin
+            width: parent.width - x * 2
+
+            //: %1 is the site that wants load popup
+            //% "We have prevented %1 from opening pop-up window. Would you like to able this website open pop-ups?"
```
Comment (llewelld) [inline apps/shared/PopupBlockedDialog.qml:41]: A suggested re-wording:
"The site %1 has requested to open a new tab. Would you like to allow it to open new tabs?"

File: apps/shared/PopupBlockedDialog.qml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            color: Theme.highlightColor
+        }
+
+        TextSwitch {
+            id: remember
+
+            //: Remember decision for this site for later use
+            //% "Remember for this site"
+            text: qsTrId("sailfish_browser-remember_for_site")
+        }
```
Comment (llewelld) [inline apps/shared/PopupBlockedDialog.qml:54]: I wonder if something like the following should also be added?
"You can change this setting in future by selecting the padlock icon in the toolbar, or from the Permissions page in Settings"

File: apps/shared/PopupBlockedDialog.qml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    onAccepted: {
+        if (rememberValue) {
+            PermissionManager.add(dialog.host, "popup", PermissionManager.Allow)
+        } else {
+            PermissionManager.add(dialog.host, "popup", PermissionManager.Allow, PermissionManager.Session)
```
Comment (llewelld) [inline apps/shared/PopupBlockedDialog.qml:60]: One thing I find a little confusing is that the Permissions list doesn't distinguish between "Session allow" and "Always allow" permissions. Is this something that could or should be included?

**Review by llewelld — CHANGES_REQUESTED** [review]
This is working nicely. There are a few things worth considering I've added as comments, and it looks like translation generation needs to be hooked up still as well.

File: apps/browser/qml/pages/PrivacySettingsPage.qml [hunk 4]
```diff
@@ -30,7 +30,7 @@ Page {
             PageHeader {
                 //: Privacy settings page header
                 //% "Privacy settings"
-                title: qsTrId("sailfish_browser-he-privacy-settingsf")
+                title: qsTrId("sailfish_browser-he-privacy-settings")
```
[the remark this answers is not in the document]
Comment (sledges) [reply apps/browser/qml/pages/PrivacySettingsPage.qml:33]: We have translation backup and will be able to restore, it's not uncommon. I'll keep an eye on when this should happen after this PR is merged.

[the remark this answers is not in the document]
Comment (llewelld) [reply apps/browser/qml/pages/PrivacySettingsPage.qml:33]: Great, thank you @sledges. And no need to alter your PR here in that case @atatarov.

**Review by llewelld — CHANGES_REQUESTED** [review]
One very small change requested, but this looks fine to me otherwise.

The following will also be needed to ensure lupdate gets run on the files with the new translation strings: https://github.com/sailfishos/sailfish-browser/pull/829

File: apps/shared/PopupBlockedDialog.qml [hunk 5]
```diff
@@ -0,0 +1,68 @@
+/****************************************************************************
+**
+** Copyright (c) 2021 Open Mobile Platform LLC.
+**
+****************************************************************************/
```
[the remark this answers is not in the document]
Comment (llewelld) [reply apps/shared/PopupBlockedDialog.qml:5]: Fixed with https://github.com/sailfishos/sailfish-browser/pull/829

**Review by llewelld — APPROVED** [review]
LGTM

**Comment by xlise** [comment]
Maybe add dependency embedlite-components version 1.22.29 "embed:popupblocked"

**Review by okodron — APPROVED** [review]

**Comment by atatarov** [comment]
> Maybe add dependency embedlite-components version 1.22.29 "embed:popupblocked"

It depends of sailfish-components-webview, but there is no dependencies of embedlite. It doesn't crash the browser and I think it could be missed

**Comment by xlise** [comment]
LGTM

**Review by okodron — APPROVED** [review]

**Review by llewelld — APPROVED** [review]
Still LGTM.

**Pull request merged**